### PR TITLE
[docs] Ask to specify tested configurations in PR body

### PR DESCRIPTION
**Pull request by RubenRBS**
Happy to discuss what kind of changes would be nice to have, but from conversations I've come to the conclusion that sometimes we would prefer contributors to be honest and let us know which test they did actually run. 

With that same idea, I've added the following comment:
> It's not mandatory to test every possible configuration if it's not easy to do so,
but we'd appreciate it if you let us know beforehand which ones you're sure do work.
If you don't have access to a specific platform to test your changes on,
feel free to reflect that in the validate() method.

As I'm still a noobie, the message could be actually wrong so please let me know if that's the case.

Feel free to close if it's not a good idea :)

Looks like this:
image

### CODE REVIEW

**Comment by SSE4** [comment]
I don't think it's needed to request people to test on multiple OS configurations.
perhaps, most of people have an access to just one system (one of Linux/Windows/macOS).
even if they have multiple, it's an additional (and inconvenient) burden to test on second PC or laptop (or via dual-boot, if it's possible, as it requires, hm... a reboot).
that's the point why we do have CI.
moreover, even if we have this information specified, it doesn't necessarily mean configuration matches CCI one. e.g. someone tested on Windows, okay, does it mean they used Clang, MinGW, Intel compiler or something else? that probably will introduce more mess than useful information, specially given the variety of Linux distributions and architecture - okay, i've tested on my Raspbian, but it doesn't build on CCI, so what?

**Comment by uilianries** [comment]
It's not mandatory but usually we need to ask users which OS they used when fixing or supporting another configuration which is not native in CCI, example, Android and MinGW. Usually I ask to users share a full log with the build output, so we can have some information about effect of their changes. I don't think this change will create confusion, but probably most of people will keep it unchecked as we have seen in current PRs and issues.

**Comment by prince-chrismc** [comment]
> but probably most of people will keep it unchecked as we have seen in current PRs and issues.

🤷 I have no issue with this but as long as CI is passing I am not sure it matters

***

We ask for PEP8 but then completely ignore that so this is a "less is more" type of situation for me

**Review by uilianries — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,7 +25,9 @@ To contribute follow the next steps:
    opening one to gather feedback.
 2. Make sure to [request access](docs/adding_packages/README.md#request-access) and be aware there is a [CLA](docs/CONTRIBUTOR_LICENSE_AGREEMENT.md).
 3. Get setup by following the [Developing Recipes](docs/developing_recipes_locally.md) guide and learn the basic commands.
-4. Check the [How To Add Packages](docs/adding_packages/README.md) page for the break down of ConanCenterIndex specific conventions and practices.
+4. Check the [How To Add Packages](docs/adding_packages/README.md) page for the breakdown of ConanCenterIndex specific conventions and practices.
+   Note that not testing your new package with a specific configuration is not mandatory if you're not able to, but please do let us know
+   in the PR message, and add the untested configurations to your validate() method.
```
Comment (danimtb) [inline CONTRIBUTING.md:30]:
```suggestion
   in the PR message, and add the untested configurations to your `validate()` method.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+ - [ ] Windows:
+ - [ ] Other, please specify:
+
+<!--
+This is also a good place to share with all of us **why you are submitting this PR**
+(specially if it is a new addition to ConanCenter):
+Is it a dependency of other libraries you want to package?
+Are you the author of the library?
+Thanks!
+-->
```
Comment (danimtb) [inline .github/PULL_REQUEST_TEMPLATE.md:23]: Does this part appear in the PR markdown? If this is hidden then I am happy to remove it as well as the preceding part.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -25,7 +25,9 @@ To contribute follow the next steps:
    opening one to gather feedback.
 2. Make sure to [request access](docs/adding_packages/README.md#request-access) and be aware there is a [CLA](docs/CONTRIBUTOR_LICENSE_AGREEMENT.md).
 3. Get setup by following the [Developing Recipes](docs/developing_recipes_locally.md) guide and learn the basic commands.
-4. Check the [How To Add Packages](docs/adding_packages/README.md) page for the break down of ConanCenterIndex specific conventions and practices.
+4. Check the [How To Add Packages](docs/adding_packages/README.md) page for the breakdown of ConanCenterIndex specific conventions and practices.
+   Note that not testing your new package with a specific configuration is not mandatory if you're not able to, but please do let us know
```
Comment (prince-chrismc) [inline CONTRIBUTING.md:29]: Take advantage of the github highlights 

```suggestion
   > **Note**: that not testing your new package with a specific configuration is not mandatory if you're not able to, but please do let us know
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,6 +1,26 @@
 Specify library name and version:  **lib/1.0**
 
-<!-- This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks! -->
+Changes tested in the following configurations:
```
Comment (SSE4) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: should be passive? `Changes have been tested`?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,6 +1,26 @@
...
+<!--
+It's not mandatory to test every possible configuration if it's not easy to do so,
+but we'd appreciate it if you let us know beforehand which ones you're sure do work.
+If you don't have access to a specific platform to test your changes on,
+feel free to reflect that in the validate() method.
+
+Also, please specify specific configuration information if necessary
```
Comment (SSE4) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: how do I know if it's necessary?

File: CONTRIBUTING.md (hunk 3 above)
Comment (SSE4) [inline CONTRIBUTING.md:29]: I don't understand double negative wording `not testing is not mandatory` and then `if you're not able`, could you rephrase it?
does it mean `testing is mandatory if you're able`, or what exactly?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,6 +1,26 @@
 Specify library name and version:  **lib/1.0**
 
-<!-- This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks! -->
+Changes tested in the following configurations:
+<!--
+It's not mandatory to test every possible configuration if it's not easy to do so,
+but we'd appreciate it if you let us know beforehand which ones you're sure do work.
+If you don't have access to a specific platform to test your changes on,
+feel free to reflect that in the validate() method.
```
Comment (SSE4) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: an example would be nice (or link to one)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (SSE4) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: I think it's visible if you click `edit`

**Comment by RubenRBS** [comment]
Not stale, just waiting

**Comment by SSE4** [comment]
> Not stale, just waiting

waiting for what exactly?

**Comment by RubenRBS** [comment]
Just for the right moment to discuss it, once there are less pressing things to be done after the v2 launch :)
